### DISCUSSION TITLE
I observed that the whole service of apm-application-toolkit was deleted in the subsequent version of skywalking 8.8.0. What is the new alternative version and plan?

### DISCUSSION THREAD

**Discussion by DearZh** [apache/skywalking#8859] [Q&A] [answered]
I have observed that the whole service of apm-application-toolkit has been deleted in skywalking 8.8.0 and subsequent versions. I would like to ask what is the subsequent planning for the next toolkit, which originally contains service packages such as log4j-1x, log4j-2x, logback, Kafka, opentracing and trace? What is the new alternative to this feature?

In addition, there is a small problem: since the source code of toolkit has been deleted in skywalking 8.8.0, why apm-toolkit-log4j-2 The latest jar package of is version 8.9.0？

Why should the administrator delete the questions I asked in Chinese. I hope it was an accident.  

I am Chinese, I hope I can also have friendly exchanges in GitHub in Chinese, thank you.
----------------------------------------------------------
我观察到apm-application-toolkit的整个服务已经在skywalking 8.8.0和后续版本中被删除。我想问下原来包含log4j-1x、log4j-2x、logback、Kafka、opentracing、trace等业务包的工具箱，后续计划是什么?这个特性的新替代品是什么?

另外，还有一个小问题，既然skywalking 8.8.0中已经删除了工具箱的源代码，为什么apm-toolkit-log4j-2最新的jar包是8.9.0版本?

为什么管理员要删除我用中文问的问题?我希望这是个意外。

我是中国人，我希望我也可以在GitHub使用中文友好交流，谢谢。

**Comment by wu-sheng** [comment, marked as answer]
Nothing gets removed. There was a discussion about this, https://github.com/apache/skywalking/discussions/7548

**Reply by DearZh** [reply]
Thank you. skywalking-java is a great way to do it.

As for the question that I submitted QA in Chinese and it was deleted, is it not allowed to use Chinese? Or what? 
 
Facilitate subsequent correction thank

**Reply by wu-sheng** [reply]
> As for the question that I submitted QA in Chinese and it was deleted, is it not allowed to use Chinese? Or what?

Yes, it is. We, mostly as volunteers, can't afford to support both languages. Many of the users/contributors are global.

**Reply by DearZh** [reply]
> > As for the question that I submitted QA in Chinese and it was deleted, is it not allowed to use Chinese? Or what?
> 
> Yes, it is. We, mostly as volunteers, can't afford to support both languages. Many of the users/contributors are global.

Okay, understood.

![image](https://user-images.githubusercontent.com/32116870/162965405-2cec9ca3-f1ad-4d94-badb-17c1dac037df.png)

Is there any way to get guidance and become a contributor, as shown above?

Finally, thank you for your reply. Thank you. Thank you

**Reply by wu-sheng** [reply]
About Contributor, I think you mean code contributors, right? This usually begins you are trying to use SkyWalking in a project(research or product env), then you will face issues(bug or new feature request) from time to time, then, you could begin to read codes, and try to find a solution.

Besides these, you could find a lot of issues(open or closed-but-not-resolved), and you could try to fix them.

There are many ways for you to engage with a community. Someone doesn't contribute codes, they write blogs, record videos, talk at conferences to share the project, use cases, etc. They are contributors from our perspective.

Mostly, I just want to highlight this, **Developing an open-source project, you should enjoy the coding and have fun. And be patient, it takes time, a lot of time at the beginning**.

About WeChat, we don't have. We have a QQ group(s), but only for users to discuss on their own. They are not under maintainers' radar. The reason is the same, we can't afford so much time on answering repeated questions. GitHub and mail list(the real official channel) are being watched.

**Reply by DearZh** [reply]
> About Contributor, I think you mean code contributors, right? This usually begins you are trying to use SkyWalking in a project(research or product env), then you will face issues(bug or new feature request) from time to time, then, you could begin to read codes, and try to find a solution.
> 
> Besides these, you could find a lot of issues(open or closed-but-not-resolved), and you could try to fix them.
> 
> There are many ways for you to engage with a community. Someone doesn't contribute codes, they write blogs, record videos, talk at conferences to share the project, use cases, etc. They are contributors from our perspective.
> 
> Mostly, I just want to highlight this, **Developing an open-source project, you should enjoy the coding and have fun. And be patient, it takes time, a lot of time at the beginning**.
> 
> About WeChat, we don't have. We have a QQ group(s), but only for users to discuss on their own. They are not under maintainers' radar. The reason is the same, we can't afford so much time on answering repeated questions. GitHub and mail list(the real official channel) are being watched.

Thanks for your reply. Thank you
